nearcore/src/config.rs

```diff
@@ -1293,15 +1294,16 @@ pub fn load_config(
             if let Err(e) = genesis.validate(genesis_validation) {
                 validation_errors.push_errors(e)
             };
-            if validator_signer.is_some()
+            // Before stateless validation we require validators to track all shards, see
+            // https://github.com/near/nearcore/issues/7388
+            if !checked_feature!("stable", StatelessValidationV0, PROTOCOL_VERSION)
```

Comment:
protocol version used here should be the current epoch's protocol version right? otherwise we will never go into if branch.

Reply:
The problem is that this code is run on startup while reading config.
We base here on protocol version built into binary. Whoever runs the binary build for stateless validation would not go into this branch.

Reply:
oh ok
but then this check will be effectively be disabled for the new binaries. any chance to move it to elsewhere?

Reply:
This opens the door for validators to misconfigure their nodes after the 2.0 release and before the protocol upgrade to stateless validation. This may compromise the blockchain security in that window of time. We are actually asking them to set tracked_shards to empty after the protocol upgrade and it's easy enough to mistake binary upgrade for protocol upgrade.

Reply:
Can we merge it so that canary node works fine on testnet: https://near.zulipchat.com/#narrow/stream/308695-nearone.2Fprivate/topic/stateless.20validation.20testnet.20protocol.20upgrade/near/450709462 ?
I am working on adding same check in a different place too, can be as follow-up PR.
Wdyt @tayfunelmas @wacban